DESCRIPTION.rst

```diff
@@ -51,41 +51,105 @@ This package supports:
 
 or::
 
   $ python -m pip install https://github.com/mpi4py/mpi4py/tarball/master
 
-You can also install it directly on Fedora (as well as RHEL and their
-derivatives using the EPEL software repository)
+Installing from source requires compilers and a working MPI
+implementation. The ``mpicc`` compiler wrapper is looked for on the
+executable search path (``PATH`` environment variable). Alternatively,
+you can set the ``MPICC`` environment variable to the full path or
+command corresponding to the MPI-aware C compiler.
 
-* using ``dnf`` and the ``mpich`` package on ``x86_64``::
+The **conda-forge** community provides ready-to-use binary packages
+from an ever growing collection of software libraries built around the
+multi-platform *conda* package manager. Three MPI implementations are
+available on conda-forge: Open MPI (Linux and macOS), MPICH (Linux and
+macOS), and Microsoft MPI (Windows). You can install mpi4py and your
+preferred MPI implementation using ``conda``::
 
-  $ dnf install mpi4py-mpich
+* to use MPICH do::
 
-* using ``dnf`` and the ``openmpi`` package on ``x86_64``::
+  $ conda install -c conda-forge mpi4py mpich
 
-  $ dnf install mpi4py-openmpi
+* to use Open MPI do::
 
-Please remember to load the correct module for your choosen MPI environment
+  $ conda install -c conda-forge mpi4py openmpi
 
-* for ``mpich`` package on ``x86_64`` do::
+* to use Microsoft MPI do::
 
-  $ module load mpi/mpich-x86_64
-  $ python -c "import mpi4py"
+  $ conda install -c conda-forge mpi4py msmpi
 
-* for ``openmpi`` package on ``x86_64`` do::
+MPICH and many of its derivatives are ABI-compatible. You can provide
+the package specification ``mpich=X.Y.*=external_*`` (where ``X`` and
+``Y`` are the major and minor version numbers) to request the conda
+package manager to use system-provided MPICH (or derivative)
+libraries.
 
-  $ module load mpi/openmpi-x86_64
-  $ python -c "import mpi4py"
+The ``openmpi`` package on conda-forge has built-in CUDA support, but
+it is disabled by default. To enable it, follow the instruction
+outlined during ``conda install``. Additionally, UCX support is also
+available once the ``ucx`` package is installed.
+
+On **Fedora Linux** systems (as well as **RHEL** and their derivatives
+using the EPEL software repository), you can install binary packages
+with the system package manager::
+
+* using ``dnf`` and the ``mpich`` package::
+
+  $ sudo dnf install python3-mpi4py-mpich
+
+* using ``dnf`` and the ``openmpi`` package::
+
+  $ sudo dnf install python3-mpi4py-openmpi
+
+Please remember to load the correct MPI module for your chosen MPI
+implementation
+
+* for the ``mpich`` package do::
+
+  $ module load mpi/mpich-$(arch)
+  $ python -c "from mpi4py import MPI"
+
+* for the ``openmpi`` package do::
+
+  $ module load mpi/openmpi-$(arch)
+  $ python -c "from mpi4py import MPI"
+
+On **Ubuntu Linux** and **Debian Linux** systems, binary packages are
```

Comment:
@drew-parsons Here are the `apt` notes. Please look at the next paragraph as well, about mpi4py being openmpi-only. Any suggestions?

Reply:
Of course, still in this PR.  I was looking in the main code!

The apt description is correct, and useful to identify that debian uses OpenMPI.
Could also mention docs are available,
"Local online documentation is available in the python-mpi4py-doc package."

With regards to using mpich on a Debian system. correct to note that libmpich-dev is needed. It is also possible (and "easy") to rebuild the debian package itself for mpich, just set the MPI variable in debian/rules to .mpich. That could help some library management problems, might introduce others. Someone choosing to use mpich on a Debian system would need to know what they're doing, in terms of impact on other packages.

Would it be interesting for you to enable joint installation of both openmpi and mpich builds at the same time?  The approach could be analogous to what I did with the serial and mpi builds of h5py, place specific implementations in subdirs (mpi4py/openmpi, mpi4py/mpich, also mpi4py/intelmpi etc) with toplevel __init__.py pulling in the preferred implementation. In principle that could be useful for other systems, not just Debian/Ubuntu. We can open a separate issue if we want to go down this path.

Reply:
A mayor issue I have with the Ubuntu/Debian approach is that they use `alternatives` to switch from MPICH/Open MPI. Beside some issues with parallel installation that usually lead to users having broken environments (libs from one packge, mpiexec from another package)... Are users without sudo access able to switch from one implementation to the other? IMHO, a modules-based approach as Fedora uses is a more robust solution, although I understand it may be slightly harder to use/understand for newbies. Perhaps I'm getting it wrong, in such case, I apologize. 

In any case, giving users that chance to use an alternative implementation (to sort out bugs or third-party requirements) is always good. Also remember that MPICH and derivatives are ABI-compatible, so having an MPICH build that could work with an alternate MPICH-ABI installation makes a lot of sense.